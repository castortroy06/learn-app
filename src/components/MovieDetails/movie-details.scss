@import '../../sass/bootstrap/variables';

.movie-details {
    background-color: $gray-300;
    padding: 5.625rem 0;
    color: $white;
    font-weight: 300;

    &--name {
        font-size: 2.5rem;
        line-height: 1.219;
        text-transform: uppercase;
        margin-bottom: 0.5rem;
        font-weight: 300;
    }

    &--container {
        padding-left: 1.875rem;
    }

    &--rate {
        width: 60px;
        height: 60px;
        font-size: 1.25rem;
        line-height: 1.219;
        border: 1px solid $white;
        border-radius: 50%;
        display: inline-block;
        padding: 18px 10px;
        text-align: center;
        position: relative;
        top: -10px;
        left: 25px;
    }

    &--categories {
        font-size: 0.875rem;
        line-height: normal;
        font-weight: 500;
        opacity: 0.5;
        margin-bottom: 1.875rem;
    }

    &--year-duration {
        font-size: 1.5rem;
        font-style: normal;
        color: $color-red;
        margin-bottom: 1.875rem;
    }

    &--year {
        margin-right: 3.1875rem;
    }

    &--description {
        opacity: 0.5;
    }
}
