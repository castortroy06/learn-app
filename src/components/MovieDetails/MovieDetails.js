import React, { useState } from "react";
import './movie-details.scss';

const MovieDetails = ({ imageUrl, name, releaseYear, duration, genres, description, rate, onClick }) => {

    return (
        <div className="movie-details row">
            <img className="col-4" data-testid="image" alt={name} src={`/images/movies/${imageUrl}`} />
            <div className="movie-details--container col-8">
                <h2 data-testid="name" className="movie-details--name">{name}<span className="movie-details--rate">{rate}</span></h2>
                <p data-testid="categories" className="movie-details--categories">{genres.join(', ')}</p>
                <p className="movie-details--year-duration">
                    <span data-testid="year" className="movie-details--year">{releaseYear}</span>
                    <span data-testid="duration" className="movie-details--duration">{duration}</span>
                </p>
                <p data-testid="description" className="movie-details--description">{description}</p>

            </div>
        </div>
    );
};

export default MovieDetails;
