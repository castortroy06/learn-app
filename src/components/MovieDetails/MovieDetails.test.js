import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

import MovieDetails from './MovieDetails';

describe('MovieDetails', () => {
    it('match snapshot', () => {
        const handleClick = () => {
            
        };

        const details = {
            'title': 'Pulp Fiction',
            'poster_path': 'https://image.tmdb.org/t/p/w500/bXNvzjULc9jrOVhGfjcc64uKZmZ.jpg',
            'release_date': 2004,
            'runtime': '2h 34m',
            'genres': ['Action & Adventure'],
            'overview': 'Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven together comprising of a series of funny, bizarre and uncalled-for incidents.â€”Soumitra',
            'vote_average': 8.9,
            'onClick': handleClick(),
        }

        const tree = renderer.create(<MovieDetails {...details} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
