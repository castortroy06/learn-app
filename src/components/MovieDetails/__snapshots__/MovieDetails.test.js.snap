// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails match snapshot 1`] = `
<div
  className="movie-details row"
>
  <img
    alt="Pulp Fiction"
    className="col-4"
    data-testid="image"
    src="/images/movies/pulp-fiction.png"
  />
  <div
    className="movie-details--container col-8"
  >
    <h2
      className="movie-details--name"
      data-testid="name"
    >
      Pulp Fiction
      <span
        className="movie-details--rate"
      >
        8.9
      </span>
    </h2>
    <p
      className="movie-details--categories"
      data-testid="categories"
    >
      Action & Adventure
    </p>
    <p
      className="movie-details--year-duration"
    >
      <span
        className="movie-details--year"
        data-testid="year"
      >
        2004
      </span>
      <span
        className="movie-details--duration"
        data-testid="duration"
      >
        2h 34m
      </span>
    </p>
    <p
      className="movie-details--description"
      data-testid="description"
    >
      Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven together comprising of a series of funny, bizarre and uncalled-for incidents.â€”Soumitra
    </p>
  </div>
</div>
`;
